packages/account/src/Sections/Security/TwoFactorAuthentication/__tests__/two-factor-authentication.spec.tsx

```diff
@@ -0,0 +1,243 @@
+            account_security: {
+                totp: {
+                    secret_key: 'hello123',
+                },
+            },
+        });
+        renderComponent({ store_config: store });
+
+        await waitFor(() => {
+            const helper_text = screen.getByText(
+                /If you are unable to scan the QR code, you can manually enter this code instead:/i
+            );
+            const secret_text = screen.getByText('hello123');
+            const clipboard_component = screen.getByTestId('2fa_clipboard');
+
+            expect(helper_text).toBeInTheDocument();
+            expect(secret_text).toBeInTheDocument();
+            expect(clipboard_component).toBeInTheDocument();
+        });
+    });
+
+    it('should render step-2 title for setting up 2FA', async () => {
+        renderComponent({ store_config: store });
+
+        await waitFor(() => {
+            const step_2_title = screen.getByText(/Enter the authentication code generated by your 2FA app:/i);
+            expect(step_2_title).toBeInTheDocument();
+        });
+    });
+
+    it('should render digitform component when 2FA is disabled', async () => {
+        renderComponent({ store_config: store });
+
+        await waitFor(() => {
+            const digitform = screen.getByTestId('digitform_2fa_disabled');
+            expect(digitform).toBeInTheDocument();
+        });
+    });
+
+    it('should render 2FA article component for mobile when 2FA is disabled', async () => {
+        (isMobile as jest.Mock).mockReturnValue(true);
+        (isDesktop as jest.Mock).mockReturnValue(false);
+
+        renderComponent({ store_config: store });
+
+        await waitFor(() => {
+            // const article_component = screen.getByTestId('two_fa_article_mobile');
+            const article_component = screen.getByText('Two-factor authentication (2FA)');
+            expect(article_component).toBeInTheDocument();
+        });
+    });
+
+    it('should render 2FA article component for dekstop when 2FA is disabled', async () => {
+        (isMobile as jest.Mock).mockReturnValue(false);
+        (isDesktop as jest.Mock).mockReturnValue(true);
+
+        renderComponent({ store_config: store });
+
+        await waitFor(() => {
+            // const article_component = screen.getByTestId('two_fa_article_mobile');
+            const article_component = screen.getByText('Two-factor authentication (2FA)');
+            expect(article_component).toBeInTheDocument();
+        });
+    });
+
+    it('should render TwoFactorEnabled component when 2FA is enabled', async () => {
+        const new_store_config = {
+            ...store,
+            client: {
+                ...store.client,
+                getTwoFAStatus: jest.fn().mockResolvedValue(true),
+                has_enabled_two_fa: true,
+            },
+        };
+        renderComponent({ store_config: new_store_config });
+
+        await waitFor(() => {
```

Comment:
do we need to await here

Reply:
Yes, since getDigitStatus sets Loading to false..meaning we update states eveytime we render. Also the function is async thus, used await